site/docs/contributing.md

```diff
@@ -0,0 +1,316 @@
+- Ensure your tests are passing and your code is properly linted.
+
+Don't hesitate to ask for help. We're here to support you. If you're worried about whether your PR will be accepted, please talk to us first.
+
+## Tests
+
+### Running Tests
+
+We use Jest for testing. To run the test suite:
+
+```sh
+npm test
+```
+
+To run tests in watch mode:
+
+```sh
+npm run test:watch
+```
+
+You can also run specific tests with:
+
+```sh
+npx jest [pattern]
+```
+
+### Writing Tests
+
+When writing tests, please:
+
+- Run them with the `--randomize` flag to ensure your mocks setup and teardown are not affecting other tests.
+- Check the coverage report to ensure your changes are covered.
+- Avoid adding additional logs to the console.
+
+## Linting and Formatting
+
+We use ESLint and Prettier for code linting and formatting. Before submitting a pull request, please run:
+
+```sh
+npm run format
+npm run lint
+```
+
+It's a good idea to run the lint command as `npm run lint -- --fix` to automatically fix some linting errors.
+
+## Building the Project
+
+To build the project:
+
+```sh
+npm run build
+```
+
+For continuous building during development:
+
+```sh
+npm run build:watch
+```
+
+## Contributing to the CLI
+
+### Running the CLI During Development
+
+We recommend using `npm link` to link your local `promptfoo` package to the global `promptfoo` package:
+
+```sh
+npm link
+promptfoo --help
+```
+
+Alternatively, you can run the CLI directly:
+
+```sh
+npm run local -- eval --config examples/cloudflare-ai/chat_config.yaml
+```
+
+When working on a new feature, we recommend setting up a local `promptfooconfig.yaml` that tests your feature. Think of this as an end-to-end test for your feature.
+
+Here's a simple example:
+
+```yaml
+providers:
+  - name: openai:chat:gpt-4o
+prompts:
+  - Translate "{{input}}" to {{language}}
+tests:
+  - vars:
+      input: 'Hello, world!'
+      language: 'English'
+    assert:
+      - type: new-assertion-type
+```
+
+## Adding a New Provider
+
+1. Create an implementation in `src/providers/SOME_PROVIDER_FILE`.
+2. Update `loadApiProvider` in `src/providers.ts` to load your provider via string.
+3. Add test cases in `test/providers.test.ts`:
+   - Test the actual provider implementation.
+   - Test loading the provider via a `loadApiProvider` test.
+
+## Contributing to the Web UI
+
+The web UI is written as a [Next.js](https://nextjs.org/) app. It is exported as a static site and hosted by a local express server when bundled.
+
+To run the web UI in dev mode:
+
+```sh
+npm run local:web
+```
+
+This will host the web UI at http://localhost:3000. This allows you to hack on the Next.js app quickly (with fast refresh) but does not cover all features (such as running evals from a web browser) because that relies on a separate server process.
+
+To test the entire thing end-to-end, we recommend building the entire project and linking it to promptfoo:
+
+```sh
+npm run build
+npm link
```

Comment:
I don't think we need to repeat the instructions to `npm link`, but we do want to let the reader know that they can use the build with the view server
```suggestion
npm run build
promptfoo view
```